### PR TITLE
Data Export - Improve performance

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
This PR makes changes to the data export service and job to improve its performance for larger sample numbers.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
1. Test export with large number of samples
2. Ensure tests pass

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/jobs/data_exports/create_job.rb [hunk 1]
```diff
@@ -30,11 +30,12 @@ def initialize_manifest(export_type)
     def create_sample_zip(data_export)
       new_zip_file = Tempfile.new(binmode: true)
       ZipKit::Streamer.open(new_zip_file) do |zip|
-        data_export.export_parameters['ids'].each do |sample_id|
-          sample = Sample.find(sample_id)
+        samples = Sample.includes(attachments: { file_attachment: :blob })
```
Comment (ericenns) [inline app/jobs/data_exports/create_job.rb:33]: **nitpick (blocking):** if you add `:project, ` inside the `includes` line 38 shouldn't cause another query to get the project.

File: app/services/data_exports/create_service.rb [hunk 2]
```diff
@@ -41,18 +41,15 @@ def validate_params
...
     def validate_sample_ids
-      project_ids = []
-
-      params['export_parameters']['ids'].each do |sample_id|
-        sample = Sample.find_by(id: sample_id)
-        raise DataExportCreateError, I18n.t('services.data_exports.create.invalid_sample_id') if sample.nil?
-
-        project_id = sample.project_id
-        project_ids << project_id unless project_ids.include?(project_id)
+      samples = Sample.where(id: params['export_parameters']['ids'])
```
Comment (ericenns) [inline app/services/data_exports/create_service.rb:44]: **nitpick (blocking):** at the end of this `where` you could use `.select(:id, :project_id)` as those are the only 2 attributes that you are using.

**Review by ericenns — CHANGES_REQUESTED** [review]
This is looking better and will make this more responsive as number of sample ids passed increases. I do have a few suggestions that will make it even more efficient.

File: app/services/data_exports/create_service.rb (hunk 2 above)
Comment (ChrisHuynh333) [reply app/services/data_exports/create_service.rb]: I get a crash when using `samples = Sample.where(id: params['export_parameters']['ids']).select(:id, :project_id)`. It seems happy when using either or. Am I missing something having the two arguments?

Comment (ChrisHuynh333) [reply app/services/data_exports/create_service.rb:44]: Made the changes as we discussed in fe2ede75f9d4240f8b71eefb3d4dce58624b27ed

File: app/jobs/data_exports/create_job.rb (hunk 1 above)
Comment (ChrisHuynh333) [reply app/jobs/data_exports/create_job.rb]: Fixed in fe2ede75f9d4240f8b71eefb3d4dce58624b27ed

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**